tools/gulp/tasks/examples.ts

```diff
@@ -0,0 +1,245 @@
+import {task} from 'gulp';
+import {sync as glob} from 'glob';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+import {buildConfig} from 'material2-build-tools';
+const {packagesDir} = buildConfig;
+
+interface ExampleMetadata {
+  component: string;
+  sourcePath: string;
+  id: string;
+  title: string;
+  additionalComponents: string[];
+  additionalFiles: string[];
+  selectorName: string[];
+}
+
+interface ParsedMetadata {
+  primary: boolean;
+  component: string;
+  title: string;
+  templateUrl: string;
+}
+
+interface ParsedMetadataResults {
+  primaryComponent: ParsedMetadata;
+  secondaryComponents: ParsedMetadata[];
+}
+
+/** Path to find the examples */
+const examplesPath = path.join(packagesDir, 'material-examples');
+
+/** Output path of the module that is being created */
+const outputModuleFilename = path.join(examplesPath, 'example-module.ts');
+
+/**
+ * Build ecmascript module import statements
+ */
+function buildImportsTemplate(metadata: ExampleMetadata): string {
+  const components = [metadata.component];
+
+  if (metadata.additionalComponents) {
+    components.push(...metadata.additionalComponents);
+  }
+
+  // imports the template from the /src/material-examples directory
+  const relativeSrcPath = path
+    .relative('./src/material-examples', metadata.sourcePath)
+    .replace('.ts', '');
+
+  return `import {${components.join(',')}} from './${relativeSrcPath}';
+`;
+}
+
+/**
+ * Builds the examples metadata including title, component, etc.
+ */
+function buildExamplesTemplate(metadata: ExampleMetadata): string {
+  // if no additional files or selectors were provided,
+  // return undefined since we don't care about if these were not found
+  const additionalFiles = metadata.additionalFiles ?
+    JSON.stringify(metadata.additionalFiles) : 'undefined';
+
+  const selectorName = metadata.selectorName ?
+    `'${metadata.selectorName.join(', ')}'` : 'undefined';
+
+  return `'${metadata.id}': {
+    title: '${metadata.title}',
+    component: ${metadata.component},
+    additionalFiles: ${additionalFiles},
+    selectorName: ${selectorName}
+  },
+  `;
+}
+
+/**
+ * Build the list of components template
+ */
+function buildListTemplate(metadata: ExampleMetadata): string {
+  const components = [metadata.component];
+
+  if (metadata.additionalComponents) {
+    components.push(...metadata.additionalComponents);
+  }
```

Comment:
This would be the same comment as for the imports. We should be able to drop the if check here.